src/view/App.tsx

```diff
@@ -9,6 +9,14 @@ import { Tab } from "./components/tab/Tab";
 import { DEVICE_LIST_KEY } from "./constants";
 import { Device } from "./container/device/Device";
 
+interface vscode {
+    postMessage(message: any): void;
+}
+declare const vscode: vscode;
+
+const sendMessage = (type: string, state: any) => {
```

Comment:
I'd abstract this out as util somewhere and use it all the places

Reply:
Will do